### PR TITLE
Test Fused MOE with padded tokens

### PR DESCRIPTION
**Pull request by vthumbe1503**
# Description

Fixes # (issue)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

Please list the changes introduced in this PR:

- Change A
- Change B

# Checklist:

- [ ] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [ ] The functionality is complete
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/pytorch/test_fusible_ops.py [hunk 1]
```diff
@@ -3703,8 +3707,8 @@ def test_grouped_mlp_cuda_graph_safe_mxfp8(
         split_sizes = [split_alignment * (i + 1) for i in range(group_size)]
         random.shuffle(split_sizes)
         split_sizes = torch.tensor(split_sizes, dtype=torch.int64, device=device)
-        in_shape = (split_sizes.sum().item(), hidden_size)
-
+        # Pad the input tokens to validate the sync-free MOE
```
Comment (timmoon10) [inline tests/pytorch/test_fusible_ops.py]: Isn't this case relevant even with CPU splits? I don't think comment is providing value.

File: tests/pytorch/test_fusible_ops.py [hunk 2]
```diff
@@ -3686,8 +3686,12 @@ def test_grouped_mlp_cuda_graph_safe_mxfp8(
         hidden_size: int = 256,
         split_alignment: int = 256,
         glu_interleave_size: int = 32,
+        token_padding: int = 2048,
     ) -> None:
-        """Grouped MLP forward+backward should be CUDA graph capturable (MXFP8)."""
+        """Grouped MLP forward+backward should be CUDA graph capturable (MXFP8).
+        Uses a padded token buffer (token_padding extra rows) to also cover
+        the sync-free MoE scenario where offsets[-1] < buffer_rows.
+        """
```
Comment (timmoon10) [inline tests/pytorch/test_fusible_ops.py]: Nit: `token_padding` is just another arg and it's about as intuitive as everything else. This docstring has weird priorities for focusing on token padding while ignoring much more important options like `activation` or `single_grouped_weight`.
```suggestion
        """Grouped MLP forward+backward should be CUDA graph capturable (MXFP8)."""
```

File: tests/pytorch/test_fusible_ops.py [hunk 3]
```diff
@@ -3703,8 +3707,8 @@ def test_grouped_mlp_cuda_graph_safe_mxfp8(
         split_sizes = [split_alignment * (i + 1) for i in range(group_size)]
         random.shuffle(split_sizes)
         split_sizes = torch.tensor(split_sizes, dtype=torch.int64, device=device)
-        in_shape = (split_sizes.sum().item(), hidden_size)
-
+        # Pad the input tokens to validate the sync-free MOE
+        in_shape = (split_sizes.sum().item() + token_padding, hidden_size)
```
Comment (timmoon10) [inline tests/pytorch/test_fusible_ops.py:3708]: Nit: Isn't this case relevant even with CPU splits?
```suggestion

        # Concatenate input tokens and pad
        in_shape = (split_sizes.sum().item() + token_padding, hidden_size)
 
```

**Review by timmoon10 — DISMISSED** [review]
LGTM, although I have very minor nits about the comments.

File: tests/pytorch/test_fusible_ops.py (hunk 3 above)
Comment (vthumbe1503) [reply tests/pytorch/test_fusible_ops.py:3708]: This isnt supported in case of CPU splits. Currently for that case we assume input shape[0] is exactly same as sum of split sizes. 
In case of BF16 we use torch.split with the split sizes which wont work if inputs are padded.
I am not exactly sure about split_quantize. I would assume even that would expect the same.

**Review by timmoon10 — APPROVED** [review]
